### PR TITLE
AO-20279-Update-NPM-Build-Test-Setup-9

### PR DESCRIPTION
**Pull request by ronilan**
This Pull Request is the **9th** and hopefully the real last one in a series of pull requests to update the build setup and manage dependencies in package.json in preparation for release.

First commit (0aefb39) updates a service package to address Dependabot concerns.

Second (13a8627) and third (b31c9f0) commits handled issues specific to the `tedious` probe, MSSQL server and Document workflow results. Test now fail when they should and pass when they can. Tedious package support is two versions back from latest (there is a need and ticket for a probe update).
 
Fourth (3bf1909) and fifth (3276d10) commits handled issues specific to the `restify` probe tests and Document workflow results. 

Last commit updated CHANGELOG.md with missing data from past releases.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -161,11 +161,33 @@ Bug fix
...
+
+Bug fix
+- pass options to koa constructor
+- check if module needs patching before doing builtin lookup
+- set txname when not set by frameworks
+
+### v10.0.1
+
+Bug fix
+- avoid triggering non-existent property warning during module.exports on Node 14
```
Comment (cheempz) [inline CHANGELOG.md:193]: Thank you for keeping the changelog updated :)

File: test/probes/tedious.test.js [hunk 2]
```diff
@@ -207,9 +215,16 @@ describe('probes.tedious ' + pkg.version, function () {
...
+    connection.on('connect', function (err) {
+      if(err) {
+        throw err
+      }
       connection.execSql(fn())
+      connection.close()
     })
+
+    // api changed between versions
+    if (semver.gte(pkg.version, '10.0.0')) connection.connect()
```
Comment (cheempz) [inline test/probes/tedious.test.js:227]: Meaning an explicit `connect()` call is needed for newer version of tedious, right?

**Review by cheempz — APPROVED** [review]
LGTM, thanks @ronilan!

File: test/probes/tedious.test.js (hunk 2 above)
Comment (ronilan) [reply test/probes/tedious.test.js:227]: Yep. Temp stopgap. We'll have to revisit again for 11.x and 12.x (AO-20490) due to other issues.

**Pull request merged**